core/test/ragtime/core_test.clj

```diff
@@ -57,6 +65,29 @@
     (is (nil? (:y @(:data database))))
     (is (= (:z @(:data database)) 3))))
 
+(deftest test-migrate-all-interrupted
+  "Tests that migrate-all stops processing migrations when the thread is 
+   interrupted."
+  (let [database (in-memory-db)
+        assoc-x  (assoc-migration "assoc-x" :x 1)
+        assoc-y  (assoc-migration "assoc-y" :y 2)
+        migrations [assoc-x (interrupted-migration) assoc-y]
+        exc (volatile! nil)]
+    (doto (Thread. (fn []
+                     (try
+                       (migrate-all database {} migrations {:strategy strategy/rebase})
+                       (catch InterruptedException e
+                         (vreset! exc e)))))
+      (.start)
+      (.join))
+    (is (= ["assoc-x" "interrupted-migration"] (:migrations @(:data database)))
+        "Migrations up to and including the one when interrupt happened are processed")
+    (is (= 1 (:x @(:data database))) "Migration before interrupt is processed fully")
+    (is (= true (:interrupted? @(:data database))) "Migration before interrupt is processed fully")
```

Comment:
Lines here exceed 80 characters in length. See the contributing document and the Clojure style guide.